### PR TITLE
Cisco meraki api

### PR DESCRIPTION
**Pull request by richlilly2004**
Fixes #

## Proposed Changes
Parsers for securityevents and configurationchanges from Cisco Meraki API
Playbooks for pulling from API w/ API Key and OrgID
Workbook

### CODE REVIEW

**Review by sreedharande — DISMISSED** [review]
Reviewed playbooks 
Get-MerakiData-OrgSecurityEvents; 
Get-MerakiData-configurationChanges

**Comment by preetikr** [comment]
@v-jayakal - please review the workbook - thanks  
@shainw and @sreedharande - please review parsers and logic apps playbooks respectively - thanks.

**Comment by v-admahe** [comment]
@richlilly2004 Could you please provide us the sample data file for this two  custom table which we are using in the parsers
MerakiConfigurationChanges_CL and MerakiSecurityEvents_CL?

Thanks,
Aditi Maheshwari

File: Workbooks/CiscoMeraki.json [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+        "comparison": "isEqualTo",
+        "value": "ConfigurationChanges"
+      },
+      "name": "query - 35"
+    }
+  ],
+  "fallbackResourceIds": [
+    ""
+  ],
+  "fromTemplateId": "sentinel-UserWorkbook",
```
Comment (v-maudan) [inline Workbooks/CiscoMeraki.json:464]: Can you please change the value of fromTemplateId in workbook to "sentinel-CiscoMeraki" as it should be sentinel-workbookname. Please refer the documentation here https://github.com/Azure/Azure-Sentinel/blob/45310fe6be3cb30cbd8f073443a0c2f9144115fa/Workbooks/README.md

**Comment by v-admahe** [comment]
@richlilly2004 Could you please provide us the sample data file for this two custom table which we are using in the parsers
MerakiConfigurationChanges_CL and MerakiSecurityEvents_CL?

Thanks,
Aditi Maheshwari

**Comment by richlilly2004** [comment]
> @richlilly2004 Could you please provide us the sample data file for this two custom table which we are using in the parsers
> MerakiConfigurationChanges_CL and MerakiSecurityEvents_CL?
> 
> Thanks,
> Aditi Maheshwari

[SampleMerakiCL.zip](https://github.com/Azure/Azure-Sentinel/files/5960083/SampleMerakiCL.zip)

**Review by richlilly2004 — COMMENTED** [review]
Added From Template

File: Workbooks/CiscoMeraki.json (hunk 1 above)
Comment (richlilly2004) [reply Workbooks/CiscoMeraki.json:464]: Added and committed

**Comment by v-admahe** [comment]
> > @richlilly2004 Could you please provide us the sample data file for this two custom table which we are using in the parsers
> > MerakiConfigurationChanges_CL and MerakiSecurityEvents_CL?
> > Thanks,
> > Aditi Maheshwari
> 
> [SampleMerakiCL.zip](https://github.com/Azure/Azure-Sentinel/files/5960083/SampleMerakiCL.zip)

Could you please provide data in json format.

Thanks,
Aditi Maheshwari

**Comment by richlilly2004** [comment]
> > > @richlilly2004 Could you please provide us the sample data file for this two custom table which we are using in the parsers
> > > MerakiConfigurationChanges_CL and MerakiSecurityEvents_CL?
> > > Thanks,
> > > Aditi Maheshwari
> > 
> > 
> > [SampleMerakiCL.zip](https://github.com/Azure/Azure-Sentinel/files/5960083/SampleMerakiCL.zip)
> 
> Could you please provide data in json format.
> 
> Thanks,
> Aditi Maheshwari
securityEvents
{
        "ts": 1610172781.654293,
        "deviceMac": "ac:17:c8:21:11:50",
        "clientMac": "c2:ef:0a:b8:13:06",
        "srcIp": "10.135.100.16:50101",
        "destIp": "10.184.19.6:80",
        "protocol": "tcp/ip",
        "priority": "1",
        "classification": "33",
        "blocked": true,
        "message": "POLICY-OTHER Adobe ColdFusion admin interface access attempt",
        "signature": "1:25975:3",
        "sigSource": "ids-vrt-balanced",
        "ruleId": "meraki:intrusion/snort/GID/1/SID/25975"
    }

ConfigurationChanges
{
        "ts": "2021-02-09T19:21:01.599762Z",
        "adminName": "Redacted",
        "adminEmail": "Redacted@contoso.com",
        "adminId": "636696397319559485",
        "networkName": "Redacted",
        "networkId": "L_636696397319513507",
        "page": "Add devices",
        "label": "Added devices",
        "oldValue": "[]",
        "newValue": "[\"Q2PN-C7T8-7EX4\"]"
    }

**Comment by v-admahe** [comment]
> > > > @richlilly2004 Could you please provide us the sample data file for this two custom table which we are using in the parsers
> > > > MerakiConfigurationChanges_CL and MerakiSecurityEvents_CL?
> > > > Thanks,
> > > > Aditi Maheshwari
> > > 
> > > 
> > > [SampleMerakiCL.zip](https://github.com/Azure/Azure-Sentinel/files/5960083/SampleMerakiCL.zip)
> > 
> > 
> > Could you please provide data in json format.
> > Thanks,
> > Aditi Maheshwari
> > securityEvents
> > {
> > "ts": 1610172781.654293,
> > "deviceMac": "ac:17:c8:21:11:50",
> > "clientMac": "c2:ef:0a:b8:13:06",
> > "srcIp": "10.135.100.16:50101",
> > "destIp": "10.184.19.6:80",
> > "protocol": "tcp/ip",
> > "priority": "1",
> > "classification": "33",
> > "blocked": true,
> > "message": "POLICY-OTHER Adobe ColdFusion admin interface access attempt",
> > "signature": "1:25975:3",
> > "sigSource": "ids-vrt-balanced",
> > "ruleId": "meraki:intrusion/snort/GID/1/SID/25975"
> > }
> 
> ConfigurationChanges
> {
> "ts": "2021-02-09T19:21:01.599762Z",
> "adminName": "Redacted",
> "adminEmail": "[Redacted@contoso.com](mailto:Redacted@contoso.com)",
> "adminId": "636696397319559485",
> "networkName": "Redacted",
> "networkId": "L_636696397319513507",
> "page": "Add devices",
> "label": "Added devices",
> "oldValue": "[]",
> "newValue": "["Q2PN-C7T8-7EX4"]"
> }

@richlilly2004 : In ConfigurationChanges sample data two columns were missing so we have added those columns (ssidName,, ssidNumber) from the data you have provided us in the excel file previously 

In the parser MerakiConfigurationChanges.txt function alias name was mention as  **CiscoMerakiConfigurationChanges** but in the workbook query you use the function name  **MerakiConfigurationChanges**. Could you please change the function name in the workbook queries as CiscoMerakiConfigurationChanges.

Thanks,
Aditi Maheshwari

File: Workbooks/CiscoMeraki.json (hunk 1 above)
Comment (v-admahe) [reply Workbooks/CiscoMeraki.json:464]: > Can you please change the value of fromTemplateId in workbook to "sentinel-CiscoMeraki" as it should be sentinel-workbookname. Please refer the documentation here https://github.com/Azure/Azure-Sentinel/blob/45310fe6be3cb30cbd8f073443a0c2f9144115fa/Workbooks/README.md

@richlilly2004 Please update the value of  fromTemplateId in workbook to "sentinel-CiscoMeraki", you have by mistakenly updated them in playbooks.

Thanks,
Aditi Maheshwari

**Comment by richlilly2004** [comment]
> CiscoMerakiConfigurationChanges

Updated

**Comment by v-admahe** [comment]
@richlilly2004  The workbookmeta file is missing in the PR. Could you please do the required changes in workbookMetadata file and add in the PR.

File: Workbooks/CiscoMeraki.json (hunk 1 above)
Comment (v-admahe) [reply Workbooks/CiscoMeraki.json:464]: @richlilly2004 Please update the value of fromTemplateId in workbook to "sentinel-CiscoMeraki", you have by mistakenly updated them in playbooks.

Thanks,
Aditi Maheshwari

**Comment by v-admahe** [comment]
@richlilly2004 : sample data file should be a part of PR. So could you please add that file in check in.

Thanks,
Aditi Maheshwari

**Review by richlilly2004 — COMMENTED** [review]
Changed naming standard

File: Workbooks/CiscoMeraki.json (hunk 1 above)
Comment (v-admahe) [reply Workbooks/CiscoMeraki.json:464]: @richlilly2004  Every thing related to workbook looks good , only fromTemplateId  value should be updated to sentinel-CiscoMeraki for approval.

could you please update the value of fromTemplateId in workbook?

Thanks,
Aditi Maheshwari

T

**Review by richlilly2004 — COMMENTED** [review]
Updated fromTemplateId

**Comment by v-jayakal** [comment]
@richlilly2004 - A suggestion, Can you add text/note that workbook is dependent on the Parsers. I am approving the PE with suggestion.
@shainw, Can you please review the Parsers in the PR?

**Review by v-jayakal — DISMISSED** [review]

**Review by richlilly2004 — COMMENTED** [review]
Added prereq comment to workbook

**Comment by richlilly2004** [comment]
> @richlilly2004 - A suggestion, Can you add text/note that workbook is dependent on the Parsers. I am approving the PE with suggestion.
> @shainw, Can you please review the Parsers in the PR?

Wasn't sure if you wanted the text in the Workbook or a readme. I tossed at the top of the workbook.

**Review by v-jayakal — DISMISSED** [review]

File: Parsers/CiscoMeraki/MerakiConfigurationChanges.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+// 1. Open Log Analytics/Azure Sentinel Logs blade. Copy the query below and paste into the Logs query window. 
+// 2. Click the Save button above the query. A pane will appear on the right, select "as Function" from the drop down. Enter a Function Name.
+//    It is recommended to name the Function Alias, as CiscoMerakiConfigurationChanges
+// 3. Kusto Functions can typically take up to 15 minutes to activate. You can then use Function Alias for other queries.
+//
+// REFERENCES: 
+// Using functions in Azure monitor log queries: https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+// 
+MerakiConfigurationChanges_CL
+| project Timestamp=ts_t,adminName=adminName_s,adminEmail=adminEmail_s,adminId=adminId_s,page=page_s,label=label_s,oldValue=oldValue_s,newValue=newValue_s,networkName=networkName_s,networkId=networkId_s,ssidName=ssidName_s, ssidNumber=ssidNumber_d
```
Comment (shainw) [inline Parsers/CiscoMeraki/MerakiConfigurationChanges.txt:20]: You are not projecting TimeGenerated here, but your sample data implies you have this value for ConfigChanges, please include that.

File: Parsers/CiscoMeraki/MerakiConfigurationChanges.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+//  
+// DESCRIPTION:
+// This parser takes raw Cisco Meraki logs from the ConfigurationChanges Cloud API and parses the logs into a normalized schema.
+//
+// USAGE:
+// 1. Open Log Analytics/Azure Sentinel Logs blade. Copy the query below and paste into the Logs query window. 
+// 2. Click the Save button above the query. A pane will appear on the right, select "as Function" from the drop down. Enter a Function Name.
+//    It is recommended to name the Function Alias, as CiscoMerakiConfigurationChanges
+// 3. Kusto Functions can typically take up to 15 minutes to activate. You can then use Function Alias for other queries.
+//
```
Comment (shainw) [inline Parsers/CiscoMeraki/MerakiConfigurationChanges.txt:15]: Please include inline Log Sample event so others understand the type of data that should be included.  See the other Meraki parser as an example - https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/CiscoMeraki/CiscoMeraki.txt

File: Parsers/CiscoMeraki/MerakiSecurityEvents.txt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+// DESCRIPTION:
+// This parser takes raw Cisco Meraki logs from the securityEvents Cloud API and parses the logs into a normalized schema.
+//
+// USAGE:
+// 1. Open Log Analytics/Azure Sentinel Logs blade. Copy the query below and paste into the Logs query window. 
+// 2. Click the Save button above the query. A pane will appear on the right, select "as Function" from the drop down. Enter a Function Name.
+//    It is recommended to name the Function Alias, as MerakiSecurityEvents
+// 3. Kusto Functions can typically take up to 15 minutes to activate. You can then use Function Alias for other queries.
+//
+// REFERENCES:
```
Comment (shainw) [inline Parsers/CiscoMeraki/MerakiSecurityEvents.txt:16]: Please include inline Log Sample event so others understand the type of data that should be included.  See the other Meraki parser as an example - https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/CiscoMeraki/CiscoMeraki.txt

**Review by shainw — CHANGES_REQUESTED** [review]
Some quick fixes for the parser, then I will approve.

**Review by richlilly2004 — COMMENTED** [review]
Updated parsers with examples and TimeGenerated

**Review by richlilly2004 — COMMENTED** [review]
Updated w/ port details in parser and workbook

**Comment by richlilly2004** [comment]
Checking in here

**Review by sreedharande — CHANGES_REQUESTED** [review]
@richlilly2004 - can you please replace your forked branch name under "Deploy to Azure" and "Deploy to Azure Gov" button Links on both README.md

**Review by richlilly2004 — COMMENTED** [review]
Updated to Azure/Azure-Sentinel repo for links

**Comment by richlilly2004** [comment]
> @richlilly2004 - can you please replace your forked branch name under "Deploy to Azure" and "Deploy to Azure Gov" button Links on both README.md

Done

**Comment by sreedharande** [comment]
@richlilly2004 - Thanks for updating README's. Both Playbooks throwing error while deploying. Following is the error message 
`The request content was invalid and could not be deserialized: 'Could not find member 'fromTemplateId' on object of type 'Template'. Path 'properties.template.fromTemplateId', line 16, position 23`

**Review by sreedharande — APPROVED** [review]

**Comment by sreedharande** [comment]
@richlilly2004 - Thanks for updating playbooks, able to deploy them without any issues. Approved Playbooks

**Pull request merged**